crates/sui-data-ingestion/src/workers/kv_store.rs

```diff
@@ -0,0 +1,216 @@
+        let mut items = vec![];
+        let mut seen = HashSet::new();
+        for (digest, value) in values {
+            if seen.contains(&digest) {
+                continue;
+            }
+            seen.insert(digest.clone());
+            let item = WriteRequest::builder()
+                .set_put_request(Some(
+                    PutRequest::builder()
+                        .item("digest", AttributeValue::B(Blob::new(digest)))
+                        .item("type", AttributeValue::S(Self::type_name(table)))
+                        .item(
+                            "bcs",
+                            AttributeValue::B(Blob::new(bcs::to_bytes(value.borrow())?)),
+                        )
+                        .build(),
+                ))
+                .build();
+            items.push(item);
+        }
+        if items.is_empty() {
+            return Ok(());
+        }
+        let mut backoff = ExponentialBackoff::default();
+        let mut queue: VecDeque<Vec<_>> = items.chunks(25).map(|ck| ck.to_vec()).collect();
+        while let Some(chunk) = queue.pop_front() {
+            let response = self
+                .dynamo_client
+                .batch_write_item()
+                .set_request_items(Some(HashMap::from([(
+                    self.table_name.clone(),
+                    chunk.to_vec(),
+                )])))
+                .send()
+                .await?;
+            if let Some(response) = response.unprocessed_items {
+                if let Some(unprocessed) = response.into_iter().next() {
+                    if !unprocessed.1.is_empty() {
+                        if queue.is_empty() {
+                            if let Some(duration) = backoff.next_backoff() {
+                                tokio::time::sleep(duration).await;
+                            }
+                        }
+                        queue.push_back(unprocessed.1);
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    async fn upload_blob<V: Serialize + std::marker::Send>(
+        &self,
+        key: Vec<u8>,
+        value: V,
+    ) -> anyhow::Result<()> {
+        let body = bcs::to_bytes(value.borrow())?.into();
+        self.s3_client
+            .put_object()
+            .bucket(self.bucket_name.clone())
+            .key(base64_url::encode(&key))
+            .body(body)
+            .send()
+            .await?;
+        Ok(())
+    }
+
+    fn type_name(table: KVTable) -> String {
+        match table {
+            KVTable::Transactions => "tx",
+            KVTable::Effects => "fx",
+            KVTable::Events => "ev",
+            KVTable::Objects => "ob",
+            KVTable::CheckpointSummary => "cs",
+            KVTable::TransactionToCheckpoint => "tx2c",
+        }
+        .to_string()
+    }
+}
+
+#[async_trait]
+impl Worker for KVStoreWorker {
+    async fn process_checkpoint(&self, checkpoint: CheckpointData) -> Result<()> {
```

Comment:
we do not have to do it in this PR, but we also have batch checkpoint processing mainly for backfill https://github.com/MystenLabs/sui/blob/6599250fa828165f00afe294ab72063c0869db25/crates/sui-indexer/src/framework/interface.rs#L9

Reply:
for a backfill, the same functionality can be used by increasing the concurrency to a higher value. Last time I ran it with 100 workers for the S3 backfill. Let me know if you have any concerns